publisher/publisher.go

```diff
@@ -90,13 +95,28 @@ func New(
 	if submissionTimeout == 0 {
 		submissionTimeout = time.Hour * 12
 	}
-	return &Impl{
+	pub := &Impl{
 		submissionTimeout: submissionTimeout,
 		issuerBundle:      bundle,
 		ctLogs:            logs,
 		log:               logger,
 		stats:             stats,
 		sa:                sa,
+		submissions:       make(map[string]*int64, len(logs)),
+	}
+	for _, log := range logs {
+		v := int64(0)
+		pub.submissions[log.statName] = &v
+	}
+	return pub
+}
+
+// ClearSubmissions clears the pub.submissions map used to track the
+// per log submission rates
+func (pub *Impl) ClearSubmissions() {
+	for _, v := range pub.submissions {
+		atomic.StoreInt64(v, 0)
+		time.Sleep(time.Second)
```

Comment:
Is there a reason that both the calling goroutine from `main.go` and the `ClearSubmissions()` function both sleep for `time.Second`? Could the sleep in `ClearSubmissions` between updating each log be removed or will that cause contention for the atomic store you're trying to avoid by resting between logs?